### ISSUE TITLE
WinUI3 Launches Blank Window

### ISSUE THREAD

**Issue by ElliottBrand** [dotnet/maui#1195]
### Description
When I attempt to launch a MAUI WinUI3 app, it launches in a blank window (solid purple background). No errors are displayed. I'm also seeing the same result in a Blazor MAUI sample project, as well as the sample weather app. Is there perhaps a potential conflict with some Windows setups? 

I set x64 as the solution platform, have the WinUI3 (Package) project set as the startup project, and launched with F5.

I'm running on dotnet 6.0.100-preview.4.21255.9 and maui-check says everything looks great.

### Screenshot
![winui_blank](https://user-images.githubusercontent.com/47930099/121073817-07d51800-c7a1-11eb-894e-c604c70c2b73.PNG)

**Comment by tespheakdeyesoftix** [comment]
I also have this problem. How to make it work?

**Comment by DamianDeMie** [comment]
I'm currently having this same issue as well on a new installation of Windows 10, It did use to work before I reinstalled Windows though which is the weird part.

**Comment by ElliottBrand** [comment]
For me, this issue appears to be limited to the BlazorWebView. A standard (non-blazor) Maui app loads the WinUI3 window okay, but a Maui Blazor app loads a blank WinUI3 window (Android emulator is fine). I've noticed if I wrap the BlazorWebView in a StackLayout and add a label, the label text loads, so it just seems to be ignoring the razor file

```
    <StackLayout>
        <Label Text="Label Text Here"></Label>
		 <b:BlazorWebView HostPage="wwwroot/index.html">
			<b:BlazorWebView.RootComponents>
				<b:RootComponent Selector="app" ComponentType="{x:Type local:Main}" />
			</b:BlazorWebView.RootComponents>
		</b:BlazorWebView>
    </StackLayout>
```

I'm wondering if it's possible the issue is related to the patch that's in place that has the WinUI3 (Package) referencing the WinUI3 class library, which is showing copies of the files in the main project? Maybe the Main.razor file just isn't being seen somehow??

**Comment by Eilon** [comment]
@PureWeen  does this seem like some of the layout issues you've seen before? I think I've seen something similar to this before but it can be difficult to reproduce.

**Comment by ElliottBrand** [comment]
@Eilon I've seen several others report a similar issue on forums, etc., while I troubleshooting the issue. However, now that I've upgraded to preview 5, I'm unable to even get a new default Maui Blazor app to start the WinUI project to see if I'm still having the same issue. I just receive a Visual Studio alert that says "The project doesn't know how to run the profile [ProjectNameHere].WinUI.

If you'd like some more specific details, just let me know. Or I could even make myself available for a video call to walk through my setup and potential cause.

**Comment by Eilon** [comment]
@ElliottBrand for preview 5 can you confirm that you've installed Project Reunion 0.8 as listed here: https://docs.microsoft.com/en-us/dotnet/maui/get-started/installation

> In addition, you must currently install the following Visual Studio extensions to create apps that target Windows UI Library (WinUI) 3:
> 
> - [Project Reunion (Preview)](https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftProjectReunionPreview)
> - [Single-project MSIX Packaging Tools](https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftSingleProjectMSIXPackagingTools)

The getting started page also suggests running the `maui-check` tool, which ensures that you have [almost] all the pre-reqs (we're working on improvements to have it install even more). (And eventually VS's installer will handle much of this anyway.)

**Comment by PureWeen** [comment]
If you resize the window does anything appear? If I recall correctly with P5 there's a bug with some layouts where resizing the window will trigger a layout

**Comment by ElliottBrand** [comment]
I didn't realize I needed to install those extensions, so I just installed them. Good idea about adding them to maui-check, that'll save a lot of headaches. I also went ahead and installed WebView2, since it looks like it's also required to use BlazorWebView.

It now launches a NEW WinUI project (without updating any NuGet packages) correctly and I'm able to see the application just fine. However, that's based on the 0.8.0-preview NuGet packages for ProjectReunion, ProjectReunion.Foundation, and ProjectReunion.WinUI. Once I update those to 0.8.0, the app starts crashing.

I did pull this out of the output window.

```Exception thrown: 'System.TypeLoadException' in Microsoft.Maui.Controls.dll```

At this point, line #63 of App.g.i.cs is highlighted, which looks to be the expected result when an unhandled exception occurs, and an alert pops up when I continue saying "A debugger is attached to [AppName].WinUI.exe but not configured to debug this unhandled exception. To debug this exception, detach the current debugger."

I tried launching both from the play button in the top menu bar as well as right-clicking on the WinUI project > Debug > Start New Instance.

With all this said, the original issue of only launching a blank window appears to be resolved for me with this preview release (assuming these extensions weren't required in the last release).
